server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -406,7 +407,8 @@ static boolean executeBulkItemRequest(
     }
 
     private static String[] getStoredFieldsSpec(IndexShard indexShard) {
-        if (indexShard.indexSettings().getIndexVersionCreated().onOrAfter(IndexVersions.INFERENCE_METADATA_FIELDS)) {
+        if (indexShard.indexSettings().getIndexVersionCreated().onOrAfter(IndexVersions.INFERENCE_METADATA_FIELDS)
+            && INFERENCE_METADATA_FIELDS_FEATURE_FLAG.isEnabled()) {
```

Comment:
I think it's worth adding a method in the InferenceMetadataFieldsMapper that does this check consistently?